### ISSUE TITLE
tests/performance/index.js not honoring opts

### ISSUE THREAD

**Issue by yaronyg** [pouchdb/pouchdb#3636]
In the performance tests we have four instances where new PouchDB instances are created but the user submitted opts values are not used in creating those new instances.

| File | Lines |
| --- | --- |
| tests/performance/perf.basics.js | [here](https://github.com/pouchdb/pouchdb/blob/b0b0200ee9a417233c83033538176e13fce3f4f4/tests/performance/perf.basics.js#L138) and [here](https://github.com/pouchdb/pouchdb/blob/b0b0200ee9a417233c83033538176e13fce3f4f4/tests/performance/perf.basics.js#L144) |
| tests/performance/replication-test.js | [here](https://github.com/pouchdb/pouchdb/blob/b0b0200ee9a417233c83033538176e13fce3f4f4/tests/performance/replication-test.js#L23) and [here](https://github.com/pouchdb/pouchdb/blob/b0b0200ee9a417233c83033538176e13fce3f4f4/tests/performance/replication-test.js#L30) |

This feels like a bug. It certainly bit me. In my case I am using the PouchDB perf tests as a way to test performance of node.js solution on Android and iOS. I don't have either environment set up correctly yet to run SqlLite or LevelDB (yet, this WILL be fixed!!!) so I wanted to use memdown.

The way I first implemented this was by editing performance\index.js to set opts to include {db: require("memdown")} but I got a bunch of adapter is missing errors.

I worked around the bug by instead setting PouchDB to use defaults and set db there. That worked and so I'm a happy camper. But the fact that the lines above don't take into account the submitted opts does seem kinda wrong. But it depends on how we think about opts. Are they meant to be global or not? If they are meant to be global then why bother with opts? Why not just use defaults? If they aren't meant to be global then why wouldn't they apply to something like tests/performance/replication-test.js line 30?

In looking at performance\index.js we only use opts to set the adapter. That seems like the kind of thing that should apply everywhere and argues that we should set it using defaults and just get rid of opts all together.

Any thoughts? Is this a bug or a feature?

**Comment by nolanlawson** [comment]
Side note: you can permalink lines of code by clicking on the line, and then pressing Y to make it permanent. E.g. you cite [these lines](https://github.com/pouchdb/pouchdb/blob/7a7d0310afcc6cfca0509e36d75986f6ce62d313/tests/performance/perf.basics.js#L112-L118) and [these lines](https://github.com/pouchdb/pouchdb/blob/7a7d0310afcc6cfca0509e36d75986f6ce62d313/tests/performance/replication-test.js#L23-L30).

What do you mean by "submitted opts"? In general it sounds like the solution here is to pass in any opts you need via URL params (in the browser) or environment vars (in Node).

**Comment by yaronyg** [comment]
I didn't know about the Y trick. Cool! I fixed the entry above. Also for some reason the line numbers I gave in perf.basics.js were broken but I now fixed them.

In this case the problem is that we are testing in JXCore. This is node.js. But it is node.js running on mobile. So we really don't have things like environment vars because there is no environment. So we need to set global configurations directly.

Ideally the way this would work is via opts but the lines I point to above don't use the opts that is passed in. That is why I had to finally hack [this](https://github.com/pouchdb/pouchdb/blob/b0b0200ee9a417233c83033538176e13fce3f4f4/tests/performance/index.js#L46) into [this](https://github.com/obastemur/jxcore-android-basics/blob/590e6c9fc8d856f017693709d1c046c8cfaf7375/jxcore-droid/assets/jxcore/memDownPerf/index.js#L48)

But it sucked to have to change a line in our tests just to figure out how to get a global config into place.

So what I'm trying to figure out is - should the 4 lines I mention above have used opts? Or do we need another mechanism for setting global config values?
